supervision/annotators/core.py

```diff
@@ -1128,84 +1142,156 @@ def annotate(
         ![label-annotator-example](https://media.roboflow.com/
         supervision-annotator-examples/label-annotator-example-purple.png)
         """
+
         assert isinstance(scene, np.ndarray)
-        font = cv2.FONT_HERSHEY_SIMPLEX
-        anchors_coordinates = detections.get_anchors_coordinates(
-            anchor=self.text_anchor
-        ).astype(int)
+        self._validate_labels(labels, detections)
+
+        labels = self._get_labels_text(detections, labels)
+        text_properties = self._get_text_properties(labels)
+
+        xyxy = self._calculate_label_positions(
+            detections, text_properties, self.text_anchor
+        )
+
```

Comment:
It seems to me that `_get_text_properties` and `_calculate_label_positions` should be merge into a single function that returns numpy array. You can than cut out `xyxy` portion of that numpy array and pass it to `spread_out_boxes`.

Reply:
addressed, removed my struct, merged functions (definitely a good idea!), and used numpy for passing data (feels very wrong). Tested for regressions - all is well.